dags/databricks-model-serve-sagemaker-example.py

```diff
@@ -0,0 +1,146 @@
+                else:
+                    deployed = True
+
+                return {'deployed': deployed, 'version': version_number, 'description': description}
+
+        # If no model is marked for Staging, then return "Not Available".
+        return {'deployed': 'Not Available', 'version': None, 'description': None}
+
+    model_status_result = check_model_version_status()
+
+    # Based on the output of the check_model_version_status() task determine if a new model needs to be deployed then
+    # proceed accordingly.
+    new_version_confirmation = ShortCircuitOperator(
+        task_id="new_version_confirmation",
+        python_callable=lambda deployed: not deployed,
+        op_kwargs=dict(deployed=model_status_result['deployed'])
+    )
+
+    @task
+    def deploy_model(image_url: str, sagemaker_execution_arn: str):
+        """Deploy new model to AWS Sagemaker
+
+        Uses mlflow-pyfunc image and Sagemaker Execution Role to deploy new model and endpoint.
+        """
+
+        sagemaker = SageMakerDeploymentClient(target_uri)
+
+        sagemaker.create_deployment(
+            name=deployment_name,
+            model_uri=model_uri,
+            config=dict(
+                image_url=f'{image_url}',
+                execution_role_arn=f'{sagemaker_execution_arn}'
+            )
+        )
+
+    @task()
+    def test_model_endpoint():
+        """Test sending requests to new endpoint
+
+        Uses Sagemaker Runtime API to send a request with sample data to get a prediction.
+        """
+
+        # Read in JSON string of sample data to test with.
+        df = pd.read_json(test_sample, orient='split')
+
+        # Get prediction from Endpoint with sample data.
+        sagemaker_runtime = boto3.client('sagemaker-runtime', region_name=region)
+
+        predictions = sagemaker_runtime.invoke_endpoint(
+            EndpointName=deployment_name,
+            Body=df.to_json(orient="split"),
+            ContentType='application/json; format=pandas-split'
+        )
+
+        logging.info('Predictions: ' + predictions['Body'].read().decode("ascii"))
+
+    @task
+    def mark_as_deployed(model_version: str):
+        """Tag model as deployed in MLflow Registry
+
+        If the previous tasks are successful this will add a "deployed: True" tag to the model version which was
+        deployed as part of his pipeline.
+
+        Keyword Arguments:
+        model_version -- model version that was deployed as part of this pipeline
+        """
+
+        client = MlflowClient()
+
+        client.set_model_version_tag(
+            name=model_name,
+            version=model_version,
+            key='deployed',
+            value="True"
+        )
+
+    new_version_confirmation >> deploy_model(
+        image_url="{{ var.value.mlflow_pyfunc_image_url }}",
+        sagemaker_execution_arn="{{ var.value.sagemaker_execution_arn }}"
+    ) >> test_model_endpoint() >> mark_as_deployed(model_status_result['version'])
```

Comment:
IMO this would be a little more readable if calling `deploy_model(...)` and `mark_as_deployed(...)` were set to variables first. (And maybe do the same for `test_module_endpoint()` as well.) Something like:
```python
_deploy_model = deploy_model(
    image_url="{{ var.value.mlflow_pyfunc_image_url }}",
    sagemaker_execution_arn="{{ var.value.sagemaker_execution_arn }}"

)
_test_model_endpoint = test_model_endpoint()
_mark_as_deployed = mark_as_deployed(model_status_result['version'])

new_version_confirmation >> _deploy_model >> _test_model_endpoint >> _mark_as_deployed
```